app/Filament/Resources/AppointmentResource.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace App\Filament\Resources;
+
+use App\Filament\Resources\AppointmentResource\Pages;
+use App\Filament\Resources\AppointmentResource\RelationManagers;
+use App\Models\Appointment;
+use Filament\Forms;
+use Filament\Forms\Components\DatePicker;
+use Filament\Forms\Components\Select;
+use Filament\Forms\Components\Textarea;
+use Filament\Forms\Components\TextInput;
+use Filament\Forms\Form;
+use Filament\Resources\Resource;
+use Filament\Tables;
+use Filament\Tables\Columns\TextColumn;
+use Filament\Tables\Table;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\SoftDeletingScope;
+
+class AppointmentResource extends Resource
+{
+    protected static ?string $model = Appointment::class;
+
+    protected static ?string $navigationIcon = 'heroicon-o-rectangle-stack';
+
+    public static function form(Form $form): Form
+    {
+        return $form
+            ->schema([
+                Select::make('patient_id')
+                    ->relationship('patient', 'name')
+                    ->label('Patient')
+                    ->required()
+                    ->columnSpan('full'),
+                TextInput::make('patient_first_name')
```

Comment:
When you pick a patient from the list, you don't have to display the first name and last name fields. Additionally, you can make an appointment without choosing a patient.